### PR TITLE
Initial decaffeination

### PR DESCRIPTION
**Pull request by mserranom**
Initial decaffeination of spelling module.

It replaces https://github.com/overleaf/spelling/pull/23, which showed lots of issues after node version bump and build scripts update. 

Once this initial decaffeination is merged and tested, next step is update Node version from 6 to 8/10 to support https://github.com/overleaf/error-type library. This requires removing v8 profiler, which is incompatible with node > 6.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -36,6 +36,14 @@ pipeline {
       }
     }
 
+    // should be disabled once Node version is updated to >=8
```
Comment (timothee-alby) [inline Jenkinsfile]: s/disabled/enabled/ ?

Comment (mserranom) [reply Jenkinsfile]:
```suggestion
    // should be enabled once Node version is updated to >=8
```

**Comment by mserranom** [comment]
@timothee-alby I think `/src` is a web convention, it's not enforced by build scripts and other decaff work is done using `/js` (as in https://github.com/overleaf/chat/pull/17/files)

**Comment by 40thieves** [comment]
>  I think /src is a web convention, it's not enforced by build scripts and other decaff work is done using /js (as in https://github.com/overleaf/chat/pull/17/files)

Sorry I did notice this when we were looking at chat.

To me, `app/src` is the better naming convention. That seems to be the convention in a lot of JS projects. It also avoid confusion if there is any compilation going on.

**Comment by mserranom** [comment]
ok, will move to `/src`

**Comment by 40thieves** [comment]
Could we open an issue to move chat to `src` too?

**Comment by mserranom** [comment]
The build scripts are incompatible with `/src` in tests:
https://github.com/overleaf/dev-environment/blob/master/build_scripts/v1.1.21/package.scripts.es.json#L8

**Comment by mserranom** [comment]
and also incompatible with `nodemon` from build scripts:
https://github.com/overleaf/dev-environment/blob/master/build_scripts/v1.1.21/nodemon.json

I think it's better to keep it as `/js`

**Pull request merged**